tests/test_ignore_order.py

```diff
@@ -779,3 +779,166 @@ def test_ignore_order_and_group_by(self):
         diff2 = DeepDiff(t1, t2, group_by='id', ignore_order=True)
         expected2 = {'iterable_item_added': {"root['BB']['ate'][1]": 'Brownies'}}
         assert expected2 == diff2
+
+    def test_compare_func(self):
```

Comment:
Could you please also add smaller unit tests with much smaller test objects? It will make it easier to debug in the future. The current test will be a better fit for `test_delta.py`. Perhaps we could add a new test class there something like `class TestDeltaCompareFunc` and add the tests that include delta there.